src/python/strelka/strelka.py

```diff
@@ -177,14 +178,15 @@ def scan_wrapper(self,
             raise
         except Exception as e:
             signal.alarm(0)
-            logging.exception(f'{self.name}: exception while scanning'
+            logging.exception(f'{self.name}: unhandled exception while scanning'
                               f' uid {file.uid if file else "_missing_"} (see traceback below)')
             self.flags.append('uncaught_exception')
+            self.event.update({"exception": "\n".join(traceback.format_exception(e))})
```

Comment:
My read on this change is we'll be storing the thrown exception into the log itself (rather than a general, short flag if caught - and if not caught - straight to stdout). Am I reading that correct? I like the idea so we can more easily troubleshoot, but do we want to restrict string length? Do we think any issues might arise from this?

Reply:
Smart to consider, as tracebacks can be lengthy. I can restrict the traceback size based on depth I believe. What's a reasonable size for this field given the volume of such events seems to be about 0.00005% of production.

Reply:
Here's a sample solution to limit the traceback length:

The whole stack

```
self.event.update({"exception": "\n".join(traceback.format_exception(e))})

Traceback (most recent call last):

  File \"/home/karl/strelka/src/python/strelka/strelka.py\", line 173, in scan_wrapper
    self.scan(data, file, options, expire_at)

  File \"/home/karl/strelka/src/python/strelka/scanners/scan_exception.py\", line 32, in scan
    one()

  File \"/home/karl/strelka/src/python/strelka/scanners/scan_exception.py\", line 24, in one
    two()

  File \"/home/karl/strelka/src/python/strelka/scanners/scan_exception.py\", line 27, in two
    three()

  File \"/home/karl/strelka/src/python/strelka/scanners/scan_exception.py\", line 30, in three
    raise Exception(\"Scanner Exception\")

Exception: Scanner Exception

```

Select n depth from the top of the stack

```
self.event.update({"exception": "\n".join(traceback.format_exception(e, limit=1))})

Traceback (most recent call last):

  File \"/home/karl/strelka/src/python/strelka/strelka.py\", line 173, in scan_wrapper
    self.scan(data, file, options, expire_at)

Exception: Scanner Exception

```

Select n depth from the bottom of the stack, best fit for finding the root problem while limiting depth

```
self.event.update({"exception": "\n".join(traceback.format_exception(e, limit=-1))})


Traceback (most recent call last):

  File \"/home/karl/strelka/src/python/strelka/scanners/scan_exception.py\", line 30, in three
    raise Exception(\"Scanner Exception\")

Exception: Scanner Exception

```

Reply:
Let's do -1. Get's us root without potential for an excess.

Reply:
Update pushed